mods/flowers/init.lua

```diff
@@ -166,39 +166,53 @@ minetest.register_node("flowers:mushroom_brown", {
 
 
 -- Mushroom spread and death
+-- Public function to enable override by mods
+
+function flowers.mushroom_spread(pos, node)
+	--- direct sunlight kills mushroom
+	if minetest.get_node_light(pos) == 15 then
+		minetest.remove_node(pos)
+		return
+	end
+	-- find empty area with soil/tree beneath around mushroom (step 1)
+	local positions = minetest.find_nodes_in_area_under_air(
+		{x = pos.x - 1, y = pos.y - 2, z = pos.z - 1},
+		{x = pos.x + 1, y = pos.y + 1, z = pos.z + 1},
+		{"group:soil", "group:tree"})
+	-- none found
+	if #positions == 0 then
+		return
+	end
+	-- select random position
+	local pos2 = positions[ math.random(#positions) ]
+	pos2.y = pos2.y + 1
+	-- find empty area with soil/tree beneath around mushroom (step 2)
+	positions = minetest.find_nodes_in_area_under_air(
+		{x = pos2.x - 1, y = pos2.y - 2, z = pos2.z - 1},
+		{x = pos2.x + 1, y = pos2.y + 1, z = pos2.z + 1},
+		{"group:soil", "group:tree"})
```

Comment:
when you say `this is faster` do you mean it spreads faster?

This code certainly spreads mushrooms faster, as it will, per test, try and plant a mushroom in a large area.

I'm not sure it's faster in execution. You execute two `find_nodes_in_area_under_air`, while the old code had none, and those are expensive C++ api calls. I'm thinking this is likely a lot slower than the original code.